synopsys-task/src/synopsys-task/tools-parameter.ts

```diff
@@ -69,44 +71,66 @@ export class SynopsysToolsParameter {
         polaris: {
           accesstoken: inputs.POLARIS_ACCESS_TOKEN,
           serverUrl: inputs.POLARIS_SERVER_URL,
-          application: { name: inputs.POLARIS_APPLICATION_NAME },
-          project: { name: inputs.POLARIS_PROJECT_NAME },
+          application: {},
+          project: {},
           assessment: { types: assessmentTypeArray },
-          branch: {},
         },
       },
     };
+
+    const azureRepositoryName = this.getAzureRepositoryName();
+
+    if (inputs.POLARIS_APPLICATION_NAME) {
+      polData.data.polaris.application.name = inputs.POLARIS_APPLICATION_NAME;
+    } else {
+      taskLib.debug("POLARIS_APPLICATION_NAME: " + azureRepositoryName);
+      polData.data.polaris.application.name = azureRepositoryName;
+    }
+
+    if (inputs.POLARIS_PROJECT_NAME) {
+      polData.data.polaris.project.name = inputs.POLARIS_PROJECT_NAME;
+    } else {
+      taskLib.debug("POLARIS_PROJECT_NAME: " + azureRepositoryName);
+      polData.data.polaris.project.name = azureRepositoryName;
+    }
+
     if (inputs.POLARIS_BRANCH_NAME) {
-      polData.data.polaris.branch.name = inputs.POLARIS_BRANCH_NAME;
+      polData.data.polaris.branch = { name: inputs.POLARIS_BRANCH_NAME };
     }
 
     if (inputs.POLARIS_TRIAGE) {
       polData.data.polaris.triage = inputs.POLARIS_TRIAGE;
     }
 
     if (parseToBoolean(inputs.POLARIS_PR_COMMENT_ENABLED)) {
-      console.info("Polaris PR comment is enabled");
-      if (!inputs.AZURE_TOKEN) {
-        throw new Error(
-          "Missing required azure token for pull request comment"
+      if (!isPullRequest()) {
+        taskLib.warning(
+          "Polaris PR comment is enabled but it will be ignored since it is not a PR/MR scan"
```

Comment:
we can align with the messages same as action:
Polaris PR Comment is ignored for non pull request scan
cross check with team again.

Reply:
updated accordingly